src/ucp/core/ucp_mm.c

```diff
@@ -713,31 +741,30 @@ ucs_status_t ucp_mem_map(ucp_context_h context, const ucp_mem_map_params_t *para
     uct_flags = ucp_mem_map_params2uct_flags(params);
 
     if (flags & UCP_MEM_MAP_ALLOCATE) {
-        status = ucp_memh_alloc(context, address, length, mem_type,
-                                uct_flags, "user memory", &memh);
-        if (status != UCS_OK) {
-            goto out;
-        }
+        status = ucp_memh_alloc(context, address, length, mem_type, uct_flags,
+                                "user memory", &memh);
     } else {
-        status = ucp_memh_create(context, address, length, mem_type,
-                                 UCT_ALLOC_METHOD_LAST, &memh);
-        if (status != UCS_OK) {
-            goto out;
-        }
+        status = ucp_memh_reg(context, address, length, mem_type, uct_flags,
+                              &memh);
+    }
 
-        status = ucp_memh_init_uct_reg(context, memh, uct_flags);
-        if (status != UCS_OK) {
-            goto err_free_memh;
-        }
+    if (status != UCS_OK) {
+        goto out;
+    }
+
+    if (memh->md_map == 0) {
```

Comment:
Can this happen if `context->reg_md_map[mem_type]` is not zero?
If not - can't we just fail while validating params?